mcs/restapi/admin_policies_test.go

```diff
@@ -0,0 +1,281 @@
+	minioRemovePolicyMock = func(name string) error {
+		return errors.New("error")
+	}
+	if err := removePolicy(adminClient, policyToRemove); assert.Error(err) {
+		assert.Equal("error", err.Error())
+	}
+}
+
+func TestAddPolicy(t *testing.T) {
+	assert := assert.New(t)
+	adminClient := adminClientMock{}
+	policyName := "new-policy"
+	policyDefinition := "{\"Version\":\"2012-10-17\",\"Statement\":[{\"Effect\":\"Allow\",\"Action\":[\"s3:GetBucketLocation\",\"s3:GetObject\",\"s3:ListAllMyBuckets\"],\"Resource\":[\"arn:aws:s3:::*\"]}]}"
+	minioAddPolicyMock = func(name, policy string) error {
+		return nil
+	}
+	minioGetPolicyMock = func(name string) (bytes []byte, err error) {
+		return []byte(policyDefinition), nil
+	}
+	assertPolicy := models.Policy{
+		Name: "new-policy",
+		Statements: []*models.Statement{
+			{
+				Actions:   []string{"s3:GetBucketLocation", "s3:GetObject", "s3:ListAllMyBuckets"},
+				Effect:    "Allow",
+				Resources: []string{"arn:aws:s3:::*"},
+			},
+		},
+		Version: "2012-10-17",
+	}
+	// Test-1 : addPolicy() adds a new policy
+	function := "addPolicy()"
+	policy, err := addPolicy(adminClient, policyName, policyDefinition)
+	if err != nil {
+		t.Errorf("Failed on %s:, error occurred: %s", function, err.Error())
+	}
+	assert.Equal(policy.Name, assertPolicy.Name)
+	assert.Equal(policy.Version, assertPolicy.Version)
+	assert.Equal(len(policy.Statements), len(assertPolicy.Statements))
+	// Test-2 : addPolicy() got an error while adding policy
+	minioAddPolicyMock = func(name, policy string) error {
+		return errors.New("error")
+	}
+	if _, err := addPolicy(adminClient, policyName, policyDefinition); assert.Error(err) {
+		assert.Equal("error", err.Error())
+	}
+	// Test-3 : addPolicy() got an error while retrieving policy
+	minioAddPolicyMock = func(name, policy string) error {
+		return nil
+	}
+	minioGetPolicyMock = func(name string) (bytes []byte, err error) {
+		return nil, errors.New("error")
+	}
+	if _, err := addPolicy(adminClient, policyName, policyDefinition); assert.Error(err) {
+		assert.Equal("error", err.Error())
+	}
+	// Test-4 : addPolicy() got an error while parsing policy
+	minioGetPolicyMock = func(name string) (bytes []byte, err error) {
+		return []byte("eaeaeaeae"), nil
+	}
+	if _, err := addPolicy(adminClient, policyName, policyDefinition); assert.Error(err) {
+		assert.NotEmpty(err.Error())
+	}
+}
+
+func TestSetPolicy(t *testing.T) {
+	assert := assert.New(t)
+	adminClient := adminClientMock{}
+	policyName := "readOnly"
+	entityName := "alevsk"
+	entityObject := "user"
+	minioSetPolicyMock = func(policyName, entityName string, isGroup bool) error {
+		return nil
+	}
+	// Test-1 : setPolicy() set policy to user
+	function := "setPolicy()"
+	err := setPolicy(adminClient, policyName, entityName, entityObject)
+	if err != nil {
+		t.Errorf("Failed on %s:, error occurred: %s", function, err.Error())
+	}
+	// Test-2 : setPolicy() set policy to group
+	entityObject = "group"
```

Comment:
if you change the definition of the entityObject then on testing we can use the` models.Group` for instance definition